### PR TITLE
LFXV2-324: Add meeting deletion message handler with registrant cleanup

### PR DESCRIPTION
**Pull request by andrest50**
## Summary
- Implements meeting deletion event handling via NATS messaging
- Adds automatic cleanup of all associated registrants when a meeting is deleted
- Fixes NATS key encoding issues in registrant repository

## Changes Made
- **New message handler**: `HandleMeetingDeleted` processes meeting deletion events
- **Concurrent cleanup**: Registrants are cleaned up concurrently with rate limiting (10 goroutines max)
- **Enhanced messaging**: Support for both fire-and-forget and request-reply NATS messages
- **Bug fix**: Fixed NATS key encoding in `deleteIndices` method to prevent "invalid key" errors
- **Validation**: Added empty email handling to prevent invalid NATS keys

## Message Flow
1. Meeting deletion triggers `MeetingDeletedMessage` publication
2. Service subscribes to `lfx.meetings-api.meeting_deleted` subject
3. Handler retrieves all meeting registrants
4. Concurrent goroutines delete registrants and send cleanup messages:
   - Index deletion messages for search services
   - Access removal messages for permission services

## Technical Details
- Uses semaphore pattern to limit concurrent operations
- Proper error handling with detailed logging
- Maintains data consistency during cleanup operations
- Backwards compatible with existing functionality

## Test Coverage
- Comprehensive unit tests for all scenarios
- Error handling and edge case coverage
- Integration tests for end-to-end flows
- Mock implementations for external dependencies

## Related
- Jira: https://linuxfoundation.atlassian.net/browse/LFXV2-324

🤖 Generated with [Claude Code](https://claude.ai/code)

### CODE REVIEW

File: internal/service/meeting_handlers.go [hunk 1]
```diff
@@ -104,3 +112,99 @@ func (s *MeetingsService) handleMeetingGetAttribute(ctx context.Context, msg dom
...
+	// Use a channel to collect errors from concurrent operations
+	type registrantError struct {
+		registrantUID string
+		err           error
+	}
+
+	errorChan := make(chan registrantError, len(registrants))
+	semaphore := make(chan struct{}, 10) // Limit concurrency to 10 goroutines
+
+	// Launch goroutines for concurrent registrant deletion
```
Comment (prabodhcs) [inline internal/service/meeting_handlers.go:165]: Can we use the worker pool here? What do you think?
e.g
```
numWorkers := 10
jobs := make(chan *models.Registrant, len(registrants))
results := make(chan registrantError, len(registrants))

for w := 0; w < numWorkers; w++ {
    go func() {
        for reg := range jobs {
            err := s.deleteRegistrantWithCleanup(ctx, reg, 0, true)
            results <- registrantError{registrantUID: reg.UID, err: err}
        }
    }()
}
for _, reg := range registrants {
    jobs <- reg
}
close(jobs)
```

**Review by prabodhcs — APPROVED** [review]
LGTM
